### PR TITLE
[OPS] #5672 Port native Import/Export plugin to OPS

### PR DESCRIPTION
**Pull request by defstat**

### CODE REVIEW

File: plugins/importexport/native/tests/functional/FunctionalNativeExportTestCase.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * @brief Test native OJS export.
+ */
+
+import('lib.pkp.tests.functional.plugins.importexport.FunctionalImportExportBaseTestCase');
+
+class FunctionalNativeExportTest extends FunctionalImportExportBaseTestCase {
+
+	public function testDoi() {
+		$export = $this->getXpathOnExport('NativeImportExportPlugin/exportIssue/1');
```
Comment (ajnyga) [inline plugins/importexport/native/tests/functional/FunctionalNativeExportTestCase.php]: reference to exportIssue

File: plugins/importexport/native/tests/functional/FunctionalNativeImportTestCase.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
+<?php
```
Comment (ajnyga) [inline plugins/importexport/native/tests/functional/FunctionalNativeImportTestCase.php]: How about with the tests, I think these are not ported yet?

**Comment by ajnyga** [comment]
ping @asmecher do you want to look at this as well?

File: plugins/importexport/native/filter/NativeXmlPreprintGalleyFilter.inc.php [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+		// representation proof files
+		return $representation;
+	}
+
+	/**
+	 * Process the self_file_ref node found inside the preprint_galley node.
+	 * @param $node DOMElement
+	 * @param $deployment NativeImportExportDeployment
+	 * @param $representation ArticleGalley
```
[the remark this answers is not in the document]
Comment (defstat) [reply plugins/importexport/native/filter/NativeXmlPreprintGalleyFilter.inc.php]: https://github.com/pkp/ops/blob/master/classes/article/ArticleGalley.inc.php

Is there a PreprintGalley class?

[the remark this answers is not in the document]
Comment (ajnyga) [reply plugins/importexport/native/filter/NativeXmlPreprintGalleyFilter.inc.php]: hmm yes true, there will be, but not yet. Easier to replace all instances later!

File: plugins/importexport/native/filter/PublicationNativeXmlFilter.inc.php [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+		// Add the series, if one is designated.
+		if ($sectionId = $entity->getData('sectionId')) {
+			$sectionDao = DAORegistry::getDAO('SectionDAO'); /** @var $sectionDao SectionDAO */
+			$section = $sectionDao->getById($sectionId);
+			assert(isset($section));
+			$entityNode->setAttribute('section_ref', $section->getLocalizedAbbrev());
+		}
+
+		$isPublished = $entity->getData('status') === STATUS_PUBLISHED;
+		$isPublished ? $entityNode->setAttribute('seq', (int) $entity->getData('seq')) : $entityNode->setAttribute('seq', '0');
```
[the remark this answers is not in the document]
Comment (defstat) [reply plugins/importexport/native/filter/PublicationNativeXmlFilter.inc.php]: It is a setting of the publication at OPS. (`publication_settings` table)

File: plugins/importexport/native/filter/NativeXmlPublicationFilter.inc.php [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+	 * Handle an element whose parent is the submission element.
+	 * @param $n DOMElement
+	 * @param $publication Publication
+	 */
+	function handleChildElement($n, $publication) {
+		switch ($n->tagName) {
+			case 'preprint_galley':
+				$this->parsePreprintGalley($n, $publication);
+				break;
+			case 'pages':
```
[the remark this answers is not in the document]
Comment (defstat) [reply plugins/importexport/native/filter/NativeXmlPublicationFilter.inc.php]: Removed

File: plugins/importexport/native/locale/en_US/locale.po [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+""
+
+msgid "plugins.importexport.native.error.unknownSection"
+msgstr "Unknown section {$param}"
+
+msgid "plugins.importexport.native.error.unknownUser"
+msgstr "The specified user, \"{$userName}\", does not exist."
+
+msgid "plugins.importexport.native.import.error.sectionTitleMismatch"
+msgstr "The section title \"{$section1Title}\" and the section title \"{$section2Title}\" in the \"{$issueTitle}\" issue matched the different existing server sections."
```
[the remark this answers is not in the document]
Comment (defstat) [reply plugins/importexport/native/locale/en_US/locale.po]: Removed

File: plugins/importexport/native/filter/NativeXmlPreprintGalleyFilter.inc.php [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+			$fileId = $fileNode->getAttribute('id');
+			$revisionId = $fileNode->getAttribute('revision');
+			$dbFileId = $deployment->getFileDBId($fileId, $revisionId);
+			if ($dbFileId) $addSubmissionFile = true;
+		}
+		$representation = parent::handleElement($node);
+
+		for ($n = $node->firstChild; $n !== null; $n=$n->nextSibling) if (is_a($n, 'DOMElement')) switch($n->tagName) {
+			case 'name':
+				// Labels are not localized in OJS ArticleGalleys, but we use the <name locale="....">...</name> structure.
```
[the remark this answers is not in the document]
Comment (defstat) [reply plugins/importexport/native/filter/NativeXmlPreprintGalleyFilter.inc.php]: Fixed

File: plugins/importexport/native/filter/PublicationNativeXmlFilter.inc.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (ajnyga) [reply plugins/importexport/native/filter/PublicationNativeXmlFilter.inc.php]: ok, let's leave it there then. Not a thing that should be solved with this plugin. And maybe there is a use case for it later.

File: plugins/importexport/native/tests/functional/FunctionalNativeImportTestCase.php (hunk 2 above)
Comment (defstat) [reply plugins/importexport/native/tests/functional/FunctionalNativeImportTestCase.php]: I am removing the tests for now.. They should be implemented across all contexts (OJS, OMP and OPS)

File: plugins/importexport/native/filter/PublicationNativeXmlFilter.inc.php [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+		$isPublished ? $entityNode->setAttribute('access_status', $entity->getData('accessStatus')) : $entityNode->setAttribute('access_status', '0');
+
+		$pages = $entity->getData('pages');
```
[the remark this answers is not in the document]
Comment (defstat) [reply plugins/importexport/native/filter/PublicationNativeXmlFilter.inc.php]: Removed

File: plugins/importexport/native/tests/functional/FunctionalNativeExportTestCase.php (hunk 1 above)
Comment (defstat) [reply plugins/importexport/native/tests/functional/FunctionalNativeExportTestCase.php]: I am removing the tests for now.. They should be implemented across all contexts (OJS, OMP and OPS)

File: plugins/importexport/native/locale/ar_IQ/locale.po [hunk 9]
```diff
@@ -0,0 +1,98 @@
+msgid ""
```
[the remark this answers is not in the document]
Comment (asmecher) [reply plugins/importexport/native/locale/ar_IQ/locale.po]: You can mark uncertain translations with the "fuzzy" flag in the PO file, e.g.:
```
#, fuzzy
msgid "plugins.importexport.native.displayName"
msgstr "إضافة Native XML"
```
I would recommend marking all locale keys that were changed in the English file with this flag. (That is, mark all languages other than English.)

[the remark this answers is not in the document]
Comment (asmecher) [reply plugins/importexport/native/locale/ar_IQ/locale.po]: Either that, or removing the translations outright when we know they're out of date.

[the remark this answers is not in the document]
Comment (asmecher) [reply plugins/importexport/native/locale/ar_IQ/locale.po]: For example, there are lots of keys that make explicit reference to issues -- those should definitely be removed to force re-translation. Ones you're less certain on, and would like a double-check, should get the `fuzzy` flag.

**Comment by asmecher** [comment]
@defstat, this looks good, but I worry about the amount of code duplication with OJS. Could you spend a bit of time looking at OJS, OPS, and OMP and seeing if more of the application import/export code can be pushed to the shared library?

**Comment by defstat** [comment]
Please check pkp/pkp-lib#6490

**Comment by asmecher** [comment]
Made obsolete by https://github.com/pkp/pkp-lib/issues/6490.

**Pull request closed without merging**